_pytest/config.py

```diff
@@ -972,11 +972,31 @@ def _initini(self, args):
         ns, unknown_args = self._parser.parse_known_and_unknown_args(args, namespace=self.option.copy())
         r = determine_setup(ns.inifilename, ns.file_or_dir + unknown_args, warnfunc=self.warn)
         self.rootdir, self.inifile, self.inicfg = r
-        self._parser.extra_info['rootdir'] = self.rootdir
-        self._parser.extra_info['inifile'] = self.inifile
-        self.invocation_dir = py.path.local()
         self._parser.addini('addopts', 'extra command line options', 'args')
         self._parser.addini('minversion', 'minimally required pytest version')
+        self._parser.addini('pathtype', 'path implementation to be used', default='pylib')
+
+        self.invocation_dir = self.make_path()
+        self.rootdir = self.make_path(self.rootdir)
```

Comment:
> How many other places is py.path.local exposed?

Unfortunately in quite a lot of places, for example the `fspath` attribute of test items, `config.rootdir` and `config.inifile` come to mind.

> So I assume you also looked at making py.path.local follow the __fspath__ protocol from PEP 519 and that it's not sanely doable?

That's an interesting idea, although I'm not sure if that would help us reach the ultimate goal which is to drop `py.path.local` altogether... at least that's what I thought the ultimate goal was.